code/datums/components/seethrough_mob.dm

```diff
@@ -0,0 +1,124 @@
+///A component that lets you turn your character transparent in order to see and click through yourself.
+/datum/component/seethrough_mob
+	///The atom that enables our dark magic
+	var/obj/render_source_atom
+	///The fake version of ourselves
+	var/image/trickery_image
+	///Which alpha do we animate towards?
+	var/target_alpha
+	///How long our faze in/out takes
+	var/animation_time
+	///Does this object let clicks from players its transparent to pass through it
+	var/clickthrough
+	///Is the seethrough effect currently active
+	var/is_active
+
+/datum/component/seethrough_mob/Initialize(target_alpha = 100, animation_time = 0.5 SECONDS, clickthrough = TRUE)
+	. = ..()
+
+	if(!ismob(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	src.target_alpha = target_alpha
+	src.animation_time = animation_time
+	src.clickthrough = clickthrough
+	src.is_active = FALSE
+	var/datum/action/toggle_seethrough/action = new(src)
+	action.Grant(parent)
+
+///Set up everything we need to trick the client and keep it looking normal for everyone else
+/datum/component/seethrough_mob/proc/trick_mob()
+	SIGNAL_HANDLER
+
+	var/mob/fool = parent
+	var/datum/hud/our_hud = fool.hud_used
+	for(var/atom/movable/screen/plane_master/seethrough in our_hud.get_true_plane_masters(SEETHROUGH_PLANE))
+		seethrough.unhide_plane(fool)
+
+	render_source_atom = new()
+
+	var/static/uid = 0
+	uid++
+	fool.render_target = "*transparent_bigmob[uid]"
```

Comment:
Mhm... Is there's no other way that doesn't override the mob's render target? Doing it this way is guaranteed to get into the way of many other features.
It'd be nice if we could have a middleman that uses the mob's render target as source and this value as render target.

Reply:
This whole system relies on * being the leading value in order to make the base render of the mob invisible, and ultimately there is no way around it. Potato did some thinking on it and there wasn't really an alternate solution without setting up a whole backend for handling cases like this one, and he told me ultimately I could PR it with the current solution and it would be corrected at some later time.

Reply:
Well then, at least implement a way to revert the render_target change when the seethrough is removed, please.